eth_tester/main.py

```diff
@@ -333,9 +334,18 @@ def get_transaction_receipt(self, transaction_hash):
         raw_transaction_hash = self.normalizer.normalize_inbound_transaction_hash(
             transaction_hash,
         )
+        transaction = self.get_transaction_by_hash(transaction_hash)
         raw_receipt = self.backend.get_transaction_receipt(raw_transaction_hash)
         self.validator.validate_outbound_receipt(raw_receipt)
         receipt = self.normalizer.normalize_outbound_receipt(raw_receipt)
+
+        if 'FORK_BYZANTIUM' in self.get_supported_forks():
+            byzantium_fork_block = self.get_fork_block('FORK_BYZANTIUM')
+        else:
+            byzantium_fork_block = None
+        if byzantium_fork_block is not None and byzantium_fork_block <= transaction['block_number']:
+            status = int.from_bytes(receipt.pop('state_root'), byteorder='big')
```

Comment:
Another option here is to use our conversion utility here:

```py
from eth_utils import to_int

to_int(receipt.pop('state_root'))
```

(and fail validation if `status > 1` since that is not a valid response according to [EIP 658](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-658.md#specification))